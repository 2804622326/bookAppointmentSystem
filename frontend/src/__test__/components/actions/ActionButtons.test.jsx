import React from 'react';
import { render, screen, fireEvent } from '@testing-library/react';
import '@testing-library/jest-dom';
import ActionButtons from '../../../components/actions/ActionButtons.jsx';

test('renders ActionButtons and handles click', () => {
  const handleClick = jest.fn();
  const { asFragment } = render(
    <ActionButtons title="Test Button" variant="primary" onClick={handleClick} />
  );

  const button = screen.getByRole('button', { name: /test button/i });
  expect(button).toBeInTheDocument();

  fireEvent.click(button);
  expect(handleClick).toHaveBeenCalledTimes(1);

  expect(asFragment()).toMatchSnapshot();
});
