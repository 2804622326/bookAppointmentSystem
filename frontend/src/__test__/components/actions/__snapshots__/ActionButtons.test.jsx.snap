// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders ActionButtons and handles click 1`] = `
<DocumentFragment>
  <button
    class="btn btn-primary btn-sm"
    type="button"
  >
    Test Button
  </button>
</DocumentFragment>
`;
